src/pages/index.js

```diff
@@ -2,85 +2,223 @@ import "./index.css";
+      console.log(err);
+    })
+    .finally(() => {
+      deleteConfirm.renderDelete(false);
+      deleteConfirm.close();
+    });
+
+const handleLikeClick = (card) => {
+  api
+    .toggleCardLike(card._id, card._isLiked)
+    .then((cardData) => {
+      card.getLikeStatus(cardData.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
+
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  function handleChangePicture(avatar) {
+    changePictureModal.renderLoading(true, "saving...", "save");
+    api
+      .setUserAvatar(avatar)
+      .then((data) => {
+        userInfo.setUserAvatar(data.avatar);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        changePictureModal.renderLoading(false, "saving...", "save");
+        changePictureModal.close();
+      });
+  }
+  // handleChangePicture
+);
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
-}
+changePictureModal.setEventListeners();
 
-function handleCardAddSubmit(data) {
-  console.log(data);
-  renderCard(data, cardListEl);
-  cardAddForm.reset();
-}
+profilePicture.addEventListener("click", () => {
+  changePictureModal.open();
+  pictureFormValidator.resetValidation();
+});
 
-/* listeners */
+const editProfileModal = new PopupWithForm(
+  "#profile-edit-modal",
+  (userData) => {
+    editProfileModal.renderLoading(true, "saving...", "save");
+    api
+      .saveUserData({ name: userData.title, about: userData.description })
+      .then((res) => {
+        setUserInfo(res);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        editProfileModal.renderLoading(false, "saving...", "save");
+        editProfileModal.close();
+      });
+    function setUserInfo(userData) {
+      userInfo.setUserInfo(userData);
+    }
+  }
+);
+editProfileModal.setEventListeners();
 
 profileEditButton.addEventListener("click", () => {
-  const useData = userInfo.getUserInfo();
-  nameInput.value = useData.name;
-  jobInput.value = useData.job;
-  editCardModal.open();
+  const values = userInfo.getUserInfo();
+  nameInput.value = values.name;
+  jobInput.value = values.about;
+  editProfileModal.open();
+  editFormValidator.resetValidation();
 });
 
+const previewImageModal = new PopupWithImage("#preview-image-modal");
+previewImageModal.setEventListeners();
+
+const addCardModal = new PopupWithForm("#profile-add-card", (cardData) => {
+  addCardModal.renderLoading(true, "creating", "create");
+  api
+    .addCard(cardData) // once
+    .then((res) => {
+      // here
+      renderCard(res);
+      // renderFunctionreplace it with a real function which renders card
+    })
+    .catch((err) => {
+      console.error(err); // log the error to the console
+    })
+    .finally(() => {
+      addCardModal.renderLoading(false, "creating...", "create");
+      addCardModal.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well